src/main/java/Duke.java

```diff
@@ -17,34 +17,42 @@ public static void main(String[] args) {
-                }
-                System.out.println("____________________________________________________________");
-            } else if (line.split(" ")[0].equals("mark")) {
-                int index = Integer.parseInt(line.split(" ")[1])-1;
-                list[index].setIsDone(true);
-                System.out.println("____________________________________________________________");
-                System.out.println("Nice! I've marked this task as done:");
-                System.out.println("[X] " + list[index].getDesc());
-                System.out.println("____________________________________________________________");
-            } else if (line.split(" ")[0].equals("unmark")) {
-                int index = Integer.parseInt(line.split(" ")[1])-1;
-                list[index].setIsDone(false);
-                System.out.println("____________________________________________________________");
-                System.out.println("OK, I've marked this task as not done yet:");
-                System.out.println("[ ] " + list[index].getDesc());
-                System.out.println("____________________________________________________________");
-            } else {
-                list[nextEmptyIndex] = new Task(line);
+            } else if (command.equals("list")) {
+                handleShowTaskList(list, nextEmptyIndex);
+            } else if (command.equals("mark")) {
+                int indexToMark = Integer.parseInt(line.split(" ")[1]) - 1;
+                handleMarkTask(list, indexToMark);
+            } else if (command.equals("unmark")) {
+                int indexToUnmark = Integer.parseInt(line.split(" ")[1]) - 1;
+                handleUnmarkTask(list, indexToUnmark);
+            } else if (command.equals("todo")) {
+                String parameters = input[1];
+                list[nextEmptyIndex] = new Task(parameters);
+                nextEmptyIndex++;
+                printAddedTask(list, nextEmptyIndex);
+            } else if (command.equals("deadline")) {
+                String[] deadlineParameters = input[1].split("/by");
+                String description = deadlineParameters[0].trim();
+                String deadline = deadlineParameters[1].trim();
+                list[nextEmptyIndex] = new Deadline(description, deadline);
                 nextEmptyIndex++;
+                printAddedTask(list, nextEmptyIndex);
+            } else if (command.equals("event")) {
+                String description = input[1].split("/from")[0].trim();
+                String from = input[1].split("/from")[1].split("/to")[0].trim();
+                String to = input[1].split("/from")[1].split("/to")[1].trim();
+                list[nextEmptyIndex] = new Event(description, from, to);
+                nextEmptyIndex++;
+                printAddedTask(list, nextEmptyIndex);
+            } else {
                 System.out.println("____________________________________________________________");
```

Comment:
Your line lengths are kept short (well within the soft limit of 110 chars), nice!